autoload/asyncomplete/sources/tscompletejob.vim

```diff
@@ -8,17 +11,25 @@ function! asyncomplete#sources#tscompletejob#completor(opt, ctx) abort
         return
     endif
 
-    call tscompletejob#complete_with_handler(0, '', {success, response->s:on_complete(success, response, a:opt, a:ctx)})
+    let s:request_id = s:request_id + 1
+    let s:requests[s:request_id] = { "opt": a:opt, "ctx": a:ctx }
+    call tscompletejob#complete_with_handler(0, '', {
+                \ "callback" : function('s:on_complete'),
+                \ "request_id" : s:request_id,
+                \ })
 endfunction
 
-function! s:on_complete(success, response, opt, ctx) abort
-    if !a:success
+function! s:on_complete(request_id, success, response) abort
```

Comment:
See next line, 2nd argument of callback `a:success` still indicate that a response was successful or not.